pdfgen.c

```diff
@@ -2491,24 +2523,35 @@ static int pdf_add_jpeg_data(struct pdf_doc *pdf, struct pdf_object *page,
                              const unsigned char *jpeg_data, size_t len)
 {
     struct pdf_object *obj;
+    uint32_t img_width, img_height;
 
-    obj = pdf_add_raw_jpeg_data(pdf, jpeg_data, len);
+    obj = pdf_add_raw_jpeg_data(pdf, jpeg_data, len, &img_width, &img_height);
     if (!obj)
         return pdf->errval;
 
+    const int ret = get_img_display_dimensions(
```

Comment:
Yeah, I normally declare these at the top of the function. It's a hang over from learning C pre C99. What you've got here is fine, since we already have other C99 requirements in this file anyway.